File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/main_program/fragmentHome/search/GetJsonData.java
package com.example.nightwingky.monstone.main_program.fragmentHome.search;

import android.util.Log;

import com.example.nightwingky.monstone.main_program.productList.ProductVO;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by nightwingky on 17-3-19.
 */

public class GetJsonData {

    public static List<ProductVO> getJsonData(String queryText) throws IOException, JSONException {
        List<ProductVO> mList = new ArrayList<>();
        ProductVO productVO;

        String jsonString = SearchHttp.putOut(queryText);
        String rightJson = jsonString.replaceAll("\\\\","");//去除转义斜杠
        rightJson = rightJson.replaceAll("\"\\[","[").replaceAll("]\"","]");//去除[]前后的引号
        JSONObject jsonObject = new JSONObject(rightJson);
        JSONArray jsonArray = jsonObject.getJSONArray("object");

        for (int i = 0; i < jsonArray.length(); i++) {
            jsonObject = jsonArray.getJSONObject(i);
            productVO = new ProductVO();
            productVO.setName(jsonObject.getString("name"));
            productVO.setSeller(jsonObject.getString("sellername"));
            productVO.setPrice(jsonObject.getString("price"));
            productVO.setCatalog(jsonObject.getString("catalog"));
            productVO.setReleaseTime(jsonObject.getString("releasetime"));
            productVO.setProductId(jsonObject.getString("no"));

            mList.add(productVO);
        }

        Log.d("searchList", String.valueOf(mList));
        return mList;
    }
}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/main_program/fragmentHome/search/GetJsonData.kt
package com.graduation.nightwingky.book_drifting.monstone.main_program.fragmentHome.search

import android.util.Log

import com.graduation.nightwingky.book_drifting.monstone.main_program.productList.ProductVO

import org.json.JSONException
import org.json.JSONObject

import java.io.IOException
import java.util.ArrayList

/**
 * Created by nightwingky on 17-3-19.
 */

object GetJsonData {

    @Throws(IOException::class, JSONException::class)
    fun getJsonData(queryText: String): List<ProductVO> {
        val mList = ArrayList<ProductVO>()
        var productVO: ProductVO

        val jsonString = SearchHttp.putOut(queryText)
        var rightJson = jsonString!!.replace("\\\\".toRegex(), "")//去除转义斜杠
        rightJson = rightJson.replace("\"\\[".toRegex(), "[").replace("]\"".toRegex(), "]")//去除[]前后的引号
        var jsonObject = JSONObject(rightJson)
        val jsonArray = jsonObject.getJSONArray("object")

        for (i in 0 until jsonArray.length()) {
            jsonObject = jsonArray.getJSONObject(i)
            productVO = ProductVO()
            productVO.name = jsonObject.getString("name")
            productVO.seller = jsonObject.getString("sellername")
            productVO.price = jsonObject.getString("price")
            productVO.catalog = jsonObject.getString("catalog")
            productVO.releaseTime = jsonObject.getString("releasetime")
            productVO.productId = jsonObject.getString("no")

            mList.add(productVO)
        }

        Log.d("searchList", mList.toString())
        return mList
    }
}
